drivers/hap_report/app/models/hap_report/erap.rb

```diff
@@ -0,0 +1,82 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HapReport
+  class Erap < GrdaWarehouseBase
+    belongs_to :hap_report, class_name: 'HapReport::Report'
+
+    def self.parse(report, data)
+      return true if data.nil?
+
+      data = data.force_encoding('UTF-8').delete_prefix("\xEF\xBB\xBF") # REMOVE UTF-8 encoding if present
+      content = StringIO.new(data)
+      header = CSV.parse(content.each_line.first).flatten.map { |m| m&.strip }
+      content.rewind
+      if header == CSV_FORMAT.keys
```

Comment:
I feel like this should have an else.  What happens if the headers don't match?  Can we expose some validation, save a note to the report that things weren't happy?